sensors/nxt_i2c_sensor_core.c

```diff
@@ -356,6 +356,22 @@ static int nxt_i2c_sensor_remove(struct i2c_client *client)
 
 static struct i2c_device_id nxt_i2c_sensor_id_table[];
 
+static int nxt_i2c_sensor_detect_lego_power(struct i2c_client *client)
+{
+	int ret;
+
+	/*
+	 * Confirm there is a device that satisfies the LMS2012 checks.
+	 */
+	ret = i2c_smbus_read_byte_data(client, 0x08);
+	if (ret < 0)
```

Comment:
I think this should be `if (ret != 0)`. It is possible (although very unlikely) that another device could be at this address. However, we would expect this device to have proper vendor and product id at these addresses and return something > 0.

Same with second instance below.